pgcli/main.py

```diff
@@ -238,8 +238,10 @@ def connect_dsn(self, dsn):
     def connect_uri(self, uri):
         uri = urlparse(uri)
         database = uri.path[1:]  # ignore the leading fwd slash
-        self.connect(database, uri.hostname, uri.username,
-                     uri.port, uri.password)
+        arguments = [database, uri.hostname, uri.username,
+                     uri.port, uri.password]
+        # unquote each URI part (they may be percent encoded)
+        self.connect(*list(map(lambda p: unquote(p) if p else p, arguments)))
```

Comment:
If we run unquote on the raw uri before we call urlparse(), it should make the code simpler. Would you mind running the unquote on https://github.com/dbcli/pgcli/pull/488/files#diff-38f3aeda57742756afa677ca09dcea43R239